### ISSUE TITLE
Same model trained with different TF versions has much less trainable parameters

### ISSUE THREAD

**Issue by aviaisr** [tensorflow/tensorflow#53248]
![image](https://user-images.githubusercontent.com/39811082/144064416-e751e7be-9d66-4e94-ae0f-3c495c24ffd2.png)
Hi,
I trained the same architecture with different TF versions 2.2.0 and 2.4.0 and got much less trainable parameters with 2.4.0 according to model.summary report, as you can see in the attached image. What is the reason for such a large difference?

**Comment by vulkomilev** [comment]
Maybe some of the parameters are just frozen for training

**Comment by tilakrayal** [comment]
@aviaisr ,
Could you please update TensorFlow to the latest stable version v.2.7 and let us know if you are facing the same error. Thanks!

**Comment by tilakrayal** [comment]
@aviaisr ,
In order to expedite the trouble-shooting process, could you please provide the complete code and dataset to reproduce the issue reported here.

**Comment by tilakrayal** [comment]
Also Could you please update TensorFlow to the latest stable version v.2.7 and let us know if you are facing the same error. Thanks!

**Comment by aviaisr** [comment]
I added the main training lines for each model to this notebook. Also, I described the differences between the training config. Architecture is the same.
Regarding the dataset, I trained the model on millions of images. 

https://colab.research.google.com/drive/14rghRb2-0kknbCfSnxcU3WR4mI8mL9La?authuser=1#scrollTo=TWL3gEpKIm0B

Thanks

**Comment by aviaisr** [comment]
> Maybe some of the parameters are just frozen for training

But why will it happen if I didn't change anything to freeze the weights?
I mainly just added weight decay regularization and changed the loss type, and of course the TF version.

**Comment by tilakrayal** [comment]
@aviaisr ,
I do not have access to the link you have provided. Could you please provide the required permissions to view the files.

**Comment by aviaisr** [comment]
Updated link:
https://colab.research.google.com/drive/14rghRb2-0kknbCfSnxcU3WR4mI8mL9La?usp=sharing

**Comment by aviaisr** [comment]
Hi,
I just saw the following:
**Load model + compile false**
Total params: 3,136,005
Trainable params: 911,913
Non-trainable params: 2,224,092

**Load weights**
Total params: 3,136,005
Trainable params: 3,101,673
Non-trainable params: 34,332

**Comment by jvishnuvardhan** [comment]
@aviaisr Looks like there are other differences (other than TF versions) between the two models. I couldn't run your colab.
Can you please share a standalone code to reproduce the issue? For comparing model.summary, we don't need training part of the code. Thanks!
